### PR TITLE
Encode signing output correctly

### PR DESCRIPTION
**Pull request by keefertaylor**
## Description

Fix encoding in signing output.

## Testing instructions

Run swift tests in XCode

## Types of changes

* Bug fix (non-breaking change which fixes an issue) 
## Checklist

- [ X ] Prefix PR title with `[WIP]` if necessary.
- [ X ] Add tests to cover changes as needed.
- [ X ] Update documentation as needed.

### CODE REVIEW

File: src/Tezos/Address.cpp [hunk 1]
```diff
@@ -73,7 +75,12 @@ std::string Address::forge() const {
         if (decoded.size() != 23 || !std::equal(prefix.begin(), prefix.end(), decoded.begin())) {
             throw std::invalid_argument("Invalid Address For forge");
         }
-        return "01" + TW::hex(decoded.begin() + prefix.size(), decoded.end()) + "00";
+        data.push_back(0x01);
+        append(data, decoded);
```
Comment (alejandro-isaza) [inline src/Tezos/Address.cpp]: Do you need to skip the prefix?

File: src/Tezos/BinaryCoding.h [hunk 2]
```diff
@@ -13,7 +13,7 @@
 
 using namespace TW;
 
-PublicKey parsePublicKey(const std::string& publicKey);
-std::string base58ToHex(const std::string& data, size_t prefixLength, uint8_t *prefix);
-std::string forgeBool(bool input);
-std::string forgePublicKey(PublicKey publicKey);
+int base58CheckDecodePrefix(const std::string& input, size_t prefixLength, uint8_t *prefix, uint8_t *output);
```
Comment (alejandro-isaza) [inline src/Tezos/BinaryCoding.h]: I don't think we need this.

File: src/Tezos/Forging.cpp [hunk 3]
```diff
@@ -9,50 +9,77 @@
...
+            forged.push_back(0x02);
             prefix[2] = 164;
             break;
         default:
           throw std::invalid_argument( "Invalid Prefix" );
     }
     const auto decoded = Base58::bitcoin.decodeCheck(publicKeyHash);
-    result += hex(decoded.begin() + prefixLength, decoded.end());
-    return result;
+    append(forged, decoded);
```
Comment (alejandro-isaza) [inline src/Tezos/Forging.cpp]: Skip prefix?

File: src/Tezos/Forging.cpp [hunk 4]
```diff
@@ -9,50 +9,77 @@
...
+            int b = input % 128;
+            input -= b;
+            input /= 128;
+            b += 128;
+            std::stringstream ss;
+            ss << std::hex << b;
+            result += ss.str();
+        }
+    }
+    return parse_hex(result);
```
Comment (alejandro-isaza) [inline src/Tezos/Forging.cpp]: We should avoid the roundtrip to hex here too.

File: src/Tezos/Address.cpp (hunk 1 above)
Comment (keefertaylor) [reply src/Tezos/Address.cpp]: Done.

File: src/Tezos/BinaryCoding.h (hunk 2 above)
Comment (keefertaylor) [reply src/Tezos/BinaryCoding.h]: Done.

File: src/Tezos/Forging.cpp (hunk 3 above)
Comment (keefertaylor) [reply src/Tezos/Forging.cpp]: Done

**Pull request merged**